Source/WebKit2/NetworkProcess/soup/NetworkProcessSoup.cpp

```diff
@@ -50,9 +52,12 @@ void NetworkProcess::userPreferredLanguagesChanged(const Vector<String>& languag
     SoupNetworkSession::defaultSession().setAcceptLanguages(languages);
 }
 
-void NetworkProcess::setProxies(WebCore::SessionID, const Vector<WebCore::Proxy>& proxies)
+void NetworkProcess::setProxies(WebCore::SessionID sessionID, const Vector<WebCore::Proxy>& proxies)
 {
-    SoupNetworkSession::defaultSession().setProxies(proxies);
+    if (auto *storageSession = NetworkStorageSession::storageSession(sessionID))
```

Comment:
Hi Carlos,
As Azam told this change is for Private Browser Session. 
We are seeing the latest master branch code that has modification in proxy setting (i.e. getOrCreateSoupNetworkSession() API is newly added). 

Can we have one more checking (i.e. Private Browser session check) and setting the proxies for particular session instead of setting in default session always.?